### PR TITLE
Add typescript usage test, fix typescript errors 

### LINKED ISSUE #110: Typescript definitions fail downstream when used
- webpack-plugin-serve Version: `master`

### How Do We Reproduce?

:boom:  **Working on PR** 

### Expected Behavior
Typescript users should be able to `npm add -D webpack-plugin-serve`, and use it in a TS file without the need to do anything else. 

### Actual Behavior

This is a sample of the errors as-is once a usage test is added (reproduced from my config):

```
~/p/webpack-plugin-serve ❯❯❯ npm run test:typescript                                                                                                                                                                           ✘ 127 

> webpack-plugin-serve@0.6.0 test:typescript /Users/kross/projects/webpack-plugin-serve
> tslint --project tsconfig.json && tsc

index.d.ts:5:8 - error TS7016: Could not find a declaration file for module 'http-proxy-middleware'. '/Users/kross/projects/webpack-plugin-serve/node_modules/http-proxy-middleware/index.js' implicitly has an 'any' type.
  Try `npm install @types/http-proxy-middleware` if it exists or add a new declaration (.d.ts) file containing `declare module 'http-proxy-middleware';`

5 } from 'http-proxy-middleware';
         ~~~~~~~~~~~~~~~~~~~~~~~

index.d.ts:6:22 - error TS7016: Could not find a declaration file for module 'koa'. '/Users/kross/projects/webpack-plugin-serve/node_modules/koa/lib/application.js' implicitly has an 'any' type.
  Try `npm install @types/koa` if it exists or add a new declaration (.d.ts) file containing `declare module 'koa';`

6 import * as Koa from 'koa';
                       ~~~~~

index.d.ts:28:19 - error TS7006: Parameter 'res' implicitly has an 'any' type.

28     setHeaders?: (res, path, stats) => any;
                     ~~~

index.d.ts:28:24 - error TS7006: Parameter 'path' implicitly has an 'any' type.

28     setHeaders?: (res, path, stats) => any;
                          ~~~~

index.d.ts:28:30 - error TS7006: Parameter 'stats' implicitly has an 'any' type.

28     setHeaders?: (res, path, stats) => any;
                                ~~~~~


Found 5 errors.
```

### PR DESCRIPTION
**Pull request by rosskevin**
Closes #110 

This PR contains:

- [x] bugfix
- [ ] feature
- [ ] refactor
- [ ] tests
- [ ] documentation
- [ ] metadata

### Breaking Changes?

- [ ] yes
- [x] no

If yes, please describe the breakage.

### Please Describe Your Changes
- the index.d.ts is invalid, and causes downstream ts users to be unable to build entirely, the errors cannot be ignored because the declarations are in-repo
- remove `declare module` - this is not needed in-repo - only for out of repo declarations/augmentations
- add ts usage test
- add tooling to test usage
- add needed types dependencies
- hook into `test` script to prevent future errors

None of the above changes change any current tests or runtime code.

If it isn't too much trouble, please release this ASAP since the current `master` breaks any TS build currently.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -61,10 +65,12 @@
   "devDependencies": {
     "@commitlint/cli": "^7.1.2",
     "@commitlint/config-conventional": "^7.1.2",
+    "@types/webpack": "^4.4.24",
```
Comment (PlayMa256) [inline package.json]: this shouldn't be on deps as well? Won't this create the same errors on consumers?

Comment (rosskevin) [reply package.json]: These types are not used in the index.d.ts so they do not need to be deps

File: index.d.ts [hunk 2]
```diff
@@ -1,99 +1,99 @@
-import { Url } from 'url';
+import { Url } from "url";
```
Comment (shellscape) [inline index.d.ts]: my only nitpick on this is to ask that this file also follow the conventions we've set in the repo and use single quote. after that, I think we're good to go.

**Comment by shellscape** [comment]
The tests for this are a welcomed addition and wonderful compliment to the project. I thank you!

File: index.d.ts (hunk 2 above)
Comment (rosskevin) [reply index.d.ts]: May I add a .prettierrc file? The vscode prettier did that on save.

**Comment by rosskevin** [comment]
Added prettierrc and re-saved the files.  I think it's good to go.

File: package.json (hunk 1 above)
Comment (rosskevin) [reply package.json]: webpack was referenced, so I moved that up to deps

**Comment by rosskevin** [comment]
This fixes a build-breaking change.  I'm not trying to be pushy but our build is broken. Is there any chance to get this published today?  If not I'll revert locally so our build can proceed.

**Comment by shellscape** [comment]
I just had a child born Monday morning and I'm short on time, so extra some patience will be needed. We like to get 2 or more approvals before merging. I'll be able to review the latest commit later this evening.

File: .prettierrc [hunk 3]
```diff
@@ -0,0 +1,5 @@
+printWidth: 100
+semi: true
+singleQuote: true
+trailingComma: all
+bracketSpacing: true
```
Comment (bebraw) [inline .prettierrc]: Why to override so many defaults?

Comment (bebraw) [reply .prettierrc]: Ah, I see. It's about matching previous style more or less.

**Comment by shellscape** [comment]
Finally got the chance to examine this. TSLint errors aren't being reported outside of the editor, meaning that any errors in the file aren't being caught by the npm script, and as such would always pass in the verification build step. Also removed the `.prettierrc` file - our prettier config is defined in `eslint-config-shellscape`. TSLint has it's own quote rule that will catch the quote style.

I'm working on cleaning that up now, but the build steo has to be fixed before we release.

**Comment by rosskevin** [comment]
@shellscape eslint is not prettier (and vice versa).  I have prettier integrated as a vscode plugin, and on every single OSS project I maintain/contribute to.  When I save a file, the vscode plugin will obey the prettier config in the project.  When the config is not there, it will use the default prettier config, which will get you exactly the file that you nitpicked.

If you don't include a prettier config, it will continue to give similar problems to others.  It is certainly your prerogative, but it mystifies me when e.g. one prefers fixing a trailing comma manually or with eslint fix vs automatically fixing it with prettier.  I can say there is my life before prettier, and after prettier, and it is just not even a thought anymore as I start or contribute to other projects.

This will help with the eslint/prettier integration https://github.com/prettier/eslint-config-prettier if you are interested.

**Comment by shellscape** [comment]
The language is a bit strong, but I'll indulge. We simply have different preferences. I had a rather long response written but it basically boils down to that one fact. While we're happy to include a type definition file, this is not a TypeScript project. And the tools in the repo are setup for the sort of project it actually is. To that end, DefinitelyTyped might be a better home for it.

I would never block a PR for a dangling comma, but switching all quotes from one form to another is a relatively large change that typically stands out. I'd wager that anyone looking at the diff would see as much as well. I don't agree that not including a prettier config will cause the same kinds of issues in the future because you were kind enough to setup TSLint, which is dynamic enough to catch most of the formatting issues that would raise a flag moving forward.

If you're interested in our prettier ruleset that we use in conjunction with ESLint, you can find them here: https://github.com/shellscape/eslint-config-shellscape/blob/b0bbe3bc077664a13141e8010e27ca31a2d6c9e0/.eslintrc.json#L11-L14

I thank you again for contributing a major improvement to the project.

**Comment by rosskevin** [comment]
My thought was to get a prettier config consistent with current styling, not proposing a style change...just proposing tooling that makes it easier for collaborators to fit-in.

**Comment by shellscape** [comment]
Given that TSLint alerts on error (in editor for those with editor plugins, when running `npm run lint`, and during CI) and that most editor plugins which support TSLint have a fix-on-save option, I think we've accomplished that without adding an additional prettier configuration file.

**Pull request merged**